File: src/test/java/day1/Day1Test.java
package day1;

import common.ResourceReaderKt;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class Day1Test {

    @Test
    public void task1_example() {
        final var input = "199\n" +
                "200\n" +
                "208\n" +
                "210\n" +
                "200\n" +
                "207\n" +
                "240\n" +
                "269\n" +
                "260\n" +
                "263\n";
        assertThat(Day1.task1(input)).isEqualTo("7");
    }

    @Test
    public void task2_example() {
        final var input = "199\n" +
                "200\n" +
                "208\n" +
                "210\n" +
                "200\n" +
                "207\n" +
                "240\n" +
                "269\n" +
                "260\n" +
                "263\n";
        assertThat(Day1.task2(input)).isEqualTo("5");
    }

    @Test
    public void task1_real() {
        final var input = ResourceReaderKt.readResource("/day01_input.txt");
        assertThat(Day1.task1(input)).isEqualTo("1559");
    }

    @Test
    public void task2_real() {
        final var input = ResourceReaderKt.readResource("/day01_input.txt");
        assertThat(Day1.task2(input)).isEqualTo("1600");
    }

}

File: src/test/kotlin/day01/Day1Test.kt
package day01

import common.readResource
import kotlin.test.Test
import kotlin.test.assertEquals

class Day1Test {

    @Test
    fun task1_example() {
        assertEquals(
            7,
            task1(readResource("/day01_example.txt"))
        )
    }

    @Test

    fun task2_example() {
        assertEquals(
            5,
            task2(readResource("/day01_example.txt"))
        )
    }

    @Test

    fun task1_real() {
        assertEquals(
            1559,
            task1(readResource("/day01_input.txt"))
        )
    }

    @Test

    fun task2_real() {
        assertEquals(
            1600,
            task2(readResource("/day01_input.txt"))
        )
    }
}